doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+  necessary).
+* When displaying the entries of a nested command to the user, should we display
+  a small label showing the name of the previous command? My gut says _yes_. In
+  the Proposal 1 example, pressing `ctrl+alt+e` to jump to "Split Pane..."
+  should probably show a small label that displays "Split Pane..." above the
+  list of nested commands.
+
+
+## Appendix 1: Name generation samples for `ShortcutAction`s
+
+### Current `ShortcutActions`
+
+* `CopyText`
+  - "Copy text"
+  - "Copy text as a single line"
+  - "Copy text without formatting"
+  - "Copy text as a single line without formatting"
+* `PasteText`
+  - "Paste text"
+* `OpenNewTabDropdown`
+  - "Open new tab dropdown"
+* `DuplicateTab`
+  - "Duplicate tab"
+* `NewTab`
+  - "Open a new tab, profile:{profile name}, directory:{directory}, commandline:{commandline}, title:{title}"
+* `NewWindow`
+  - "Open a new window"
+  - "Open a new window, profile:{profile name}, directory:{directory}, commandline:{commandline}, title:{title}"
+* `CloseWindow`
+  - "Close window"
+* `CloseTab`
+  - "Close tab"
+* `ClosePane`
+  - "Close pane"
+* `NextTab`
+  - "Switch to the next tab"
+* `PrevTab`
+  - "Switch to the previous tab"
+* `SplitPane`
+  - "Open a new pane, profile:{profile name}, split direction:{direction}, split size:{X%/Y chars}, resize parents, directory:{directory}, commandline:{commandline}, title:{title}"
+  - "Duplicate the current pane, split direction:{direction}, split size:{X%/Y chars}, resize parents, directory:{directory}, commandline:{commandline}, title:{title}"
+* `SwitchToTab`
+  - "Switch to tab {index}"
+* `AdjustFontSize`
+  - "Increase the font size"
+  - "Decrease the font size"
+* `ResetFontSize`
+  - "Reset the font size"
+* `ScrollUp`
+  - "Scroll up a line"
+  - "Scroll up {amount} lines"
+* `ScrollDown`
+  - "Scroll down a line"
+  - "Scroll down {amount} lines"
+* `ScrollUpPage`
+  - "Scroll up a page"
+  - "Scroll up {amount} pages"
+* `ScrollDownPage`
+  - "Scroll down a page"
+  - "Scroll down {amount} pages"
+* `ResizePane`
+  - "Resize pane {direction}"
+  - "Resize pane {direction} {percent}%"
+* `MoveFocus`
+  - "Move focus {direction}"
+* `Find`
+  - "Toggle the search box"
+* `ToggleFullscreen`
+  - "Toggle fullscreen mode"
+* `OpenSettings`
+  - "Open settings"
+  - "Open settings file"
+  - "Open default settings file"
+* `ToggleCommandPalette`
+  - "Toggle the Command Palette"
+  - "Toggle the Command Palette in commandline mode"
+
+### Other yet unimplemented actions:
+* `SwitchColorScheme`
+  - "Select color scheme {name}"
+* `ToggleRetroEffect`
+  - "Toggle the retro terminal effect"
+* `ExecuteCommandline`
+  - "Run a wt commandline: {cmdline}"
+* `ExecuteActions`
+  - OPINION: THIS ONE SHOULDN'T HAVE A NAME. We're not including any of these by
+    default. The user knows what they're putting in the settings by adding this
+    action, let them name it.
+  - Alternatively: "Run actions: {action.ToName() for action in actions}"

```

Comment:
Similar to the NewTerminalArgs stuff I mentioned before, we could try and keep it short and just do something like "Execute Actions (Custom)".

Reply:
(same point as above about the generated strings being unique to a particular ActionAndArgs value)